uber/reports.py

```diff
@@ -0,0 +1,48 @@
+from uber.common import *
+
+
```

Comment:
Yeah, in that case you could have something like ``PersonalizedBadgeReport.preprocessors`` which would be a list of callback functions, where each row is passed to each callback before being written.  The implementation would probably look something like

```python
class ReportBase:
    preprocessors = []

    def write_row(self, row, out):
        for callback in self.preprocessors:
            row = callback(row)
        out.writerow(row)
```

and then in the plugin you'd have something like

```python
uber.reports.ReportBase.preprocessors.append(lambda row: row + [generate_barcode(row[0])])
```

though for the sake of clarity you'd probably write that as a regular function instead of a lambda.